### PR TITLE
Add SAS Token Authentication Support to Azure Repo Plugin

### PR DESCRIPTION
**Pull request by original-brownbear**
* Added setting for SAS token
* Added support for the token in tests
* Relates #42117

-------

backport of #42982

### CODE REVIEW

File: plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java [hunk 1]
```diff
@@ -189,10 +194,6 @@ public String getName() {
         return name;
     }
 
-    public String getKey() {
```
Comment (original-brownbear) [inline plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java]: I removed this API here as well as adding the `key` to the `toString` output, just like it was done in `master` when SAS token support was added.

**Comment by original-brownbear** [comment]
@ywelsch can you double check this one, just because it's 6.8 and wasn't a clean backport (required a bit of moving around with the constructors of the settings class since we have some additional legacy settings in 6.8, other than that it went pretty clean now that we have the renaming 6.8).

I also manually verified that 3rd party tests still work with key as well as SAS token auth :)

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**Comment by original-brownbear** [comment]
thanks @ywelsch !

**Pull request merged**